clients/src/main/java/org/apache/kafka/common/security/ssl/SslFactory.java

```diff
@@ -327,11 +339,22 @@ KeyStore load() {
                 // If a password is not set access to the truststore is still available, but integrity checking is disabled.
                 char[] passwordChars = password != null ? password.value().toCharArray() : null;
                 ks.load(in, passwordChars);
+                fileLastModifiedMs = lastModifiedMs(path);
+
+                log.debug("Loaded key store with path {} modification time {}", path, new Date(fileLastModifiedMs));
                 return ks;
             } catch (GeneralSecurityException | IOException e) {
                 throw new KafkaException("Failed to load SSL keystore " + path + " of type " + type, e);
             }
         }
+
+        private long lastModifiedMs(String path) {
+            return new File(path).lastModified();
```

Comment:
@mimaison Thanks for the review. I think we shouldn't throw an exception if we can't get modification time since this is a configuration update where keystore/truststore configs were not updated and we want to keep existing behaviour for that case if we dont know that the file was updated. But it is a good point, we should have some logging for this case. Added a warning.

Reply:
I agree with @mimaison that we should use `Files.getLastModifiedTime`. Even if we decide to just log, we would know the reason for the failure. With the current approach, we will see a warning, but with no information on the root cause.

Reply:
In general, it's best to avoid the old File APIs that don't provide useful errors when things fail.

Reply:
@ijuma @mimaison Thanks for the reviews. I have updated to use `Files.getLastModifiedTime` and log the exception in the case of error.